src/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -637,11 +638,9 @@ function discoverController(
     $scope.minimumVisibleRows = $scope.hits;
   };
 
-  let loadingVis;
   function setupVisualization() {
```

Comment:
This method returned a Promise earlier, that would resolve once the `ready:vis` was fired. This method sets up the histogram on top of the Discover table and we assume it wanted to wait for that visualization to be ready. But since this only used the `visualization` and not the `visualize` directive, that even was never fired by the visualization.

It was actually working, because the doc_table below it sent that event too and that was the one that was causing the promise to be resolved. @weltenwort and me checked, that not returning a promise, but continue in the calling place directly won't change behavior at all.

Also the promise was resolved with `$scope.vis`. Since we never used that resolved value, I didn't bother returning that value from the method anymore at all.